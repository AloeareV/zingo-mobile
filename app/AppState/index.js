import Address from './Address';
import AddressBookEntry from './AddressBookEntry';
import AppStateLoaded from './AppStateLoaded';
import AppStateLoading from './AppStateLoading';
import ConnectedCompanionApp from './ConnectedCompanionApp';
import ErrorModalData from './ErrorModalData';
import InfoType from './InfoType';
import PasswordState from './PasswordState';
import ReceivePageState from './ReceivePageState';
import SendJsonToType from './SendJsonToType';
import SendPageState from './SendPageState';
import SendProgress from './SendProgress';
import ServerSelectState from './ServerSelectState';
import SettingsFileEntry from './SettingsFileEntry';
import SyncStatus from './SyncStatus';
import SyncStatusReport from './SyncStatusReport';
import ToAddr from './ToAddr';
import TotalBalance from './TotalBalance';
import Transaction from './Transaction';
import TxDetailType from './TxDetailType';
import WalletSeed from './WalletSeed';
import WalletSettings from './WalletSettings';

export {
  AppStateLoaded,
  AppStateLoading,
  Address,
  AddressBookEntry,
  ConnectedCompanionApp,
  ErrorModalData,
  InfoType,
  PasswordState,
  ReceivePageState,
  SendJsonToType,
  SendPageState,
  SendProgress,
  ServerSelectState,
  SettingsFileEntry,
  SyncStatus,
  SyncStatusReport,
  ToAddr,
  TotalBalance,
  Transaction,
  TxDetailType,
  WalletSeed,
  WalletSettings,
};
