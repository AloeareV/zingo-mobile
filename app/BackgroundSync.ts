import RPCModule from './RPCModule';
import AsyncStorage from '@react-native-async-storage/async-storage';
import NetInfo, { NetInfoStateType } from '@react-native-community/netinfo';
import { RPCSyncStatusType } from './rpc/types/RPCSyncStatusType';

const BackgroundSync = async (task_data: any) => {
  // this not interact with the server.
  const exists = await RPCModule.walletExists();

  // only if exists the wallet file make sense to do the sync.
  if (exists && exists !== 'false') {
    // only if we have connection make sense to call RPCModule.
    const networkState = await NetInfo.fetch();
    if (
      !networkState.isConnected ||
      networkState.type === NetInfoStateType.cellular ||
      (networkState.details !== null && networkState.details.isConnectionExpensive)
    ) {
      //console.log(
      //  'BS: Not started (connected: ' + networkState.isConnected,
      //  +', type: ' +
      //    networkState.type +
      //    +', expensive connection: ' +
      //    networkState.details?.isConnectionExpensive +
      //    ')',
      //);
      return;
    }
    // if the App goes to Foreground kill the interval
    const background = await AsyncStorage.getItem('@background');
    if (background === 'no') {
      //console.log('BS: Not started (going to foreground)');
      return;
    }

    let batch_num = -1;
    console.log('BS:', task_data);

    // finishEarly has two fields: wait, and done.
    // wait() returns a promise, which is resolved when
    // done() is called
    let finishEarly = manuallyResolve();

    let saver = setInterval(async () => {
      const networkStateSaver = await NetInfo.fetch();
      if (
        !networkStateSaver.isConnected ||
        networkStateSaver.type === NetInfoStateType.cellular ||
        (networkStateSaver.details !== null && networkStateSaver.details.isConnectionExpensive)
      ) {
        //console.log(
        //  'BS: Interrupted (connected: ' + networkStateSaver.isConnected,
        //  +', type: ' +
        //    networkStateSaver.type +
        //    +', expensive connection: ' +
        //    networkStateSaver.details?.isConnectionExpensive +
        //    ')',
        //);
        clearInterval(saver);
        finishEarly.done();
        return;
      }
      // if the App goes to Foreground kill the interval
      const backgroundSaver = await AsyncStorage.getItem('@background');
      if (backgroundSaver === 'no') {
        clearInterval(saver);
        //console.log('BS: Finished (going to foreground)');
        finishEarly.done();
        return;
      }

      const syncStatusStr: string = await RPCModule.execute('syncstatus', '');
      if (syncStatusStr) {
        if (syncStatusStr.toLowerCase().startsWith('error')) {
          //console.log(`BS: Error sync status ${syncStatusStr}`);
          return;
        }
      } else {
        //console.log('BS: Internal Error sync status');
        return;
      }
      // TODO: verify this JSON parse
      const ss: RPCSyncStatusType = await JSON.parse(syncStatusStr);

      //console.log('BS:', ss);
      if (ss.batch_num && ss.batch_num > -1 && batch_num !== ss.batch_num) {
        await RPCModule.doSave();
        //console.log('BS: saving...');
        // update batch_num with the new value, otherwise never change
        batch_num = ss.batch_num;
      }
    }, 5000);

    await Promise.race([RPCModule.execute('sync', ''), finishEarly.wait()]);
    clearInterval(saver);
  } else {
    console.log('BS: wallet file does not exist');
  }
  //console.log('BS: Finished (end of syncing)');
};

export default BackgroundSync;

function manuallyResolve() {
  let resolve: Function;
  // new Promise takes a function as an argument. When that function is called
  // the promise resolves with the value output by that function.
  // By passing the function out of the promise, we can call it later
  // in order to resolve the promise at will
  const promise = new Promise(fun => {
    resolve = fun;
  });

  function done() {
    resolve();
  }

  function wait() {
    return promise;
  }

  return { wait, done };
}
