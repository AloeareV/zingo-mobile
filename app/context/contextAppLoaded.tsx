import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import {
  AppStateLoaded,
  SyncingStatusReportClass,
  TotalBalanceClass,
  ReceivePageStateClass,
  ErrorModalDataClass,
  SendProgressClass,
  WalletSettingsClass,
  TransactionType,
  InfoType,
  SyncingStatusType,
  WalletSeedType,
  DimensionsType,
  AddressClass,
  AddressBookClass,
  zecPriceType,
  BackgroundType,
  SendPageStateClass,
  ToAddrClass,
  NetInfoType,
  BackgroundErrorType,
} from '../AppState';

export const defaultAppStateLoaded: AppStateLoaded = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  dimensions: {} as DimensionsType,
  appState: '' as string,
  netInfo: {} as NetInfoType,

  syncingStatusReport: new SyncingStatusReportClass(),
  totalBalance: new TotalBalanceClass(),
  addressPrivateKeys: new Map(),
  addresses: [] as AddressClass[],
  addressBook: [] as AddressBookClass[],
  transactions: [] as TransactionType[],
  sendPageState: new SendPageStateClass(new ToAddrClass(0)),
  receivePageState: new ReceivePageStateClass(''),
  info: {} as InfoType,
  walletSettings: new WalletSettingsClass(),
  syncingStatus: {} as SyncingStatusType,
  errorModalData: new ErrorModalDataClass('', ''),
  sendProgress: new SendProgressClass(0, 0, 0),
  walletSeed: {} as WalletSeedType,
  isMenuDrawerOpen: false,
  selectedMenuDrawerItem: '' as string,
  aboutModalVisible: false,
  computingModalVisible: false,
  settingsModalVisible: false,
  infoModalVisible: false,
  rescanModalVisible: false,
  seedViewModalVisible: false,
  seedChangeModalVisible: false,
  seedBackupModalVisible: false,
  seedServerModalVisible: false,
  syncReportModalVisible: false,
  poolsModalVisible: false,
  newServer: '' as string,
  uaAddress: '' as string,

  server: '' as string,
  currency: '',
  language: 'en',

  zecPrice: {
    zecPrice: 0,
    date: 0,
  } as zecPriceType,
  sendAll: false,
  background: {
    batches: 0,
    date: 0,
  } as BackgroundType,

  translate: () => '',
  backgroundError: {} as BackgroundErrorType,
};

export const ContextAppLoaded = React.createContext(defaultAppStateLoaded);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoaded;
};

export const ContextAppLoadedProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextAppLoaded.Provider value={value}>{children}</ContextAppLoaded.Provider>;
};
