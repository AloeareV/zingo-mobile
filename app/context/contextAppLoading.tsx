import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import {
  AppStateLoading,
  DimensionsType,
  InfoType,
  TotalBalanceClass,
  WalletSeedType,
  zecPriceType,
  backgroundType,
} from '../AppState';

export const defaultAppStateLoading: AppStateLoading = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  dimensions: {} as DimensionsType,
  appState: '' as string,

  screen: 0,
  actionButtonsDisabled: false,
  walletExists: false,
  walletSeed: {} as WalletSeedType,
  totalBalance: new TotalBalanceClass(),
  info: {} as InfoType,

  server: '' as string,
  currency: '',
  language: 'en',

  zecPrice: {
    zecPrice: 0,
    date: 0,
  } as zecPriceType,
  sendAll: false,
  background: {
    batches: 0,
    date: 0,
  } as backgroundType,

  translate: () => '',
};

export const ContextAppLoading = React.createContext(defaultAppStateLoading);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoading;
};

export const ContextAppLoadingProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextAppLoading.Provider value={value}>{children}</ContextAppLoading.Provider>;
};
