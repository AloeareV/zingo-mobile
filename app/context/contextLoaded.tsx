import React, { ReactNode } from 'react';

import {
  AppStateLoaded,
  SyncStatusReport,
  TotalBalance,
  SendPageState,
  ReceivePageState,
  ToAddr,
  ErrorModalData,
  SendProgress,
  WalletSettings,
} from '../AppState';

const defaultAppState: AppStateLoaded = {
  navigation: null,
  route: null,
  dimensions: {} as {
    width: number;
    height: number;
    orientation: 'portrait' | 'landscape';
    deviceType: 'tablet' | 'phone';
    scale: number;
  },

  syncStatusReport: new SyncStatusReport(),
  totalBalance: new TotalBalance(),
  addressPrivateKeys: new Map(),
  addresses: [],
  addressBook: [],
  transactions: null,
  sendPageState: new SendPageState(new ToAddr(0)),
  receivePageState: new ReceivePageState(),
  info: null,
  rescanning: false,
  wallet_settings: new WalletSettings(),
  syncingStatus: null,
  errorModalData: new ErrorModalData(),
  txBuildProgress: new SendProgress(),
  walletSeed: null,
  isMenuDrawerOpen: false,
  selectedMenuDrawerItem: '',
  aboutModalVisible: false,
  computingModalVisible: false,
  settingsModalVisible: false,
  infoModalVisible: false,
  rescanModalVisible: false,
  seedViewModalVisible: false,
  seedChangeModalVisible: false,
  seedBackupModalVisible: false,
  seedServerModalVisible: false,
  syncReportModalVisible: false,
  poolsModalVisible: false,
  newServer: null,
  uaAddress: null,

  translate: () => {},
  openErrorModal: () => {},
  closeErrorModal: () => {},
  toggleMenuDrawer: () => {},
  setComputingModalVisible: () => {},
  syncingStatusMoreInfoOnClick: async () => {},
  poolsMoreInfoOnClick: async () => {},
  startRescan: () => {},
};

export const ContextLoaded = React.createContext(defaultAppState);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoaded;
};

export const ContextLoadedProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextLoaded.Provider value={value}>{children}</ContextLoaded.Provider>;
};
