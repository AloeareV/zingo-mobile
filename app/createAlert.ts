import AsyncStorage from '@react-native-async-storage/async-storage';
import { Alert } from 'react-native';
import Toast from 'react-native-simple-toast';

export const createAlert = async (
  setBackgroundError: (title: string, error: string) => void,
  title: string,
  error: string,
  toast?: boolean,
) => {
  const background = await AsyncStorage.getItem('@background');
  if (background === 'yes') {
    setBackgroundError(title, error);
  } else {
    if (toast) {
      setTimeout(() => {
        Toast.show(error, Toast.LONG);
      }, 1000);
    } else {
      Alert.alert(title, error);
    }
  }
};
