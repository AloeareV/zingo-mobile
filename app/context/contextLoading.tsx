import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import { AppStateLoading, DimensionsType, InfoType, TotalBalanceClass, WalletSeedType } from '../AppState';

export const defaultAppStateLoading: AppStateLoading = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  dimensions: {} as DimensionsType,

  screen: 0,
  actionButtonsDisabled: false,
  walletExists: false,
  walletSeed: {} as WalletSeedType,
  server: '' as string,
  currency: '',
  language: 'en',
  totalBalance: new TotalBalanceClass(),
  info: {} as InfoType,

  translate: () => '',
};

export const ContextLoading = React.createContext(defaultAppStateLoading);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoading;
};

export const ContextLoadingProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextLoading.Provider value={value}>{children}</ContextLoading.Provider>;
};
