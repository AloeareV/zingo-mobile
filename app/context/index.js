import { defaultAppStateLoaded, ContextAppLoaded, ContextAppLoadedProvider } from './contextAppLoaded';
import { defaultAppStateLoading, ContextAppLoading, ContextAppLoadingProvider } from './contextAppLoading';

export {
  defaultAppStateLoaded,
  ContextAppLoaded,
  ContextAppLoadedProvider,
  defaultAppStateLoading,
  ContextAppLoading,
  ContextAppLoadingProvider,
};
