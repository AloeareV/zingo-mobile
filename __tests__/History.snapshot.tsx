/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import History from '../components/History';
import { defaultAppStateLoaded, ContextAppLoadedProvider } from '../app/context';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.', // us
      groupingSeparator: ',', // us
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('moment', () => {
  // Here we are able to mock chain builder pattern
  const mMoment = {
    format: (p: string) => {
      if (p === 'MMM YYYY') {
        return 'Dec 2022';
      } else if (p === 'YYYY MMM D h:mm a') {
        return '2022 Dec 13 8:00 am';
      } else if (p === 'MMM D, h:mm a') {
        return 'Dec 13, 8:00 am';
      }
    },
  };
  // Here we are able to mock the constructor and to modify instance methods
  const fn = () => {
    return mMoment;
  };
  // Here we are able to mock moment methods that depend on moment not on a moment instance
  fn.locale = jest.fn();
  return fn;
});
jest.mock('moment/locale/es', () => () => ({
  defineLocale: jest.fn(),
}));
jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native').View;
  return {
    TouchableOpacity: View,
  };
});

// test suite
describe('Component Transactions - test', () => {
  //snapshot test
  test('Transactions Portrait - snapshot', () => {
    const state = defaultAppStateLoaded;
    state.transactions = [
      {
        type: 'sent',
        address: 'sent-address-12345678901234567890',
        amount: 0.12345678,
        position: '',
        confirmations: 22,
        txid: 'sent-txid-1234567890',
        time: Date.now(),
        zec_price: 33.33,
        detailedTxns: [],
      },
      {
        type: 'receive',
        address: 'receive-address-12345678901234567890',
        amount: 0.87654321,
        position: '',
        confirmations: 133,
        txid: 'receive-txid-1234567890',
        time: Date.now(),
        zec_price: 66.66,
        detailedTxns: [],
      },
    ];
    state.uaAddress = 'UA-12345678901234567890';
    state.addresses = [
      {
        uaAddress: 'UA-12345678901234567890',
        address: 'UA-12345678901234567890',
        addressKind: 'u',
        containsPending: false,
        receivers: 'ozt',
      },
      {
        uaAddress: 'UA-12345678901234567890',
        address: 'sapling-12345678901234567890',
        addressKind: 'z',
        containsPending: false,
        receivers: 'z',
      },
      {
        uaAddress: 'UA-12345678901234567890',
        address: 'transparent-12345678901234567890',
        addressKind: 't',
        containsPending: false,
        receivers: 't',
      },
    ];
    state.translate = () => 'text translated';
    state.dimensions = {
      width: 200,
      height: 400,
      orientation: 'portrait',
      deviceType: 'tablet',
      scale: 1.5,
    };
    state.info.currencyName = 'ZEC';
    state.totalBalance.total = 1.12345678;
    const onFunction = jest.fn();
    const transactions = render(
      <ContextAppLoadedProvider value={state}>
        <History
          doRefresh={onFunction}
          toggleMenuDrawer={onFunction}
          poolsMoreInfoOnClick={onFunction}
          syncingStatusMoreInfoOnClick={onFunction}
          setZecPrice={onFunction}
        />
      </ContextAppLoadedProvider>,
    );
    expect(transactions.toJSON()).toMatchSnapshot();
  });
});
