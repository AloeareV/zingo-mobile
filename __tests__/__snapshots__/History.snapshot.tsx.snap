// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Transactions - test Transactions - snapshot 1`] = `
<View
  accessibilityLabel="text translated"
  accessible={true}
  style={
    {
      "display": "flex",
      "justifyContent": "flex-start",
      "marginBottom": 140,
      "width": "100%",
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "display": "flex",
        "paddingBottom": 0,
        "paddingTop": 10,
        "zIndex": -1,
      }
    }
    testID="header"
  >
    <Image
      source={
        {
          "testUri": "../../../assets/img/logobig-zingo.png",
        }
      }
      style={
        {
          "height": 80,
          "resizeMode": "contain",
          "width": 80,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "baseline",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "baseline",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 36,
                }
              }
            >
              á™‡
            </Text>
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 36,
                  "fontWeight": "700",
                }
              }
            >
               1.1234
            </Text>
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 25.2,
                  "paddingBottom": 0,
                }
              }
            >
              5678
            </Text>
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 36,
                }
              }
            >
               ZEC
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginVertical": 5,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "flexWrap": "wrap",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 18,
              "fontWeight": "600",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
          testID="transaction text"
        >
          text translated
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <
          color="red"
          icon={
            {
              "icon": [
                640,
                512,
                [
                  62337,
                  "cloud-download",
                  "cloud-download-alt",
                ],
                "f0ed",
                "M144 480C64.5 480 0 415.5 0 336c0-62.8 40.2-116.2 96.2-135.9c-.1-2.7-.2-5.4-.2-8.1c0-88.4 71.6-160 160-160c59.3 0 111 32.2 138.7 80.2C409.9 102 428.3 96 448 96c53 0 96 43 96 96c0 12.2-2.3 23.8-6.4 34.6C596 238.4 640 290.1 640 352c0 70.7-57.3 128-128 128H144zm79-167l80 80c9.4 9.4 24.6 9.4 33.9 0l80-80c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-39 39V184c0-13.3-10.7-24-24-24s-24 10.7-24 24V318.1l-39-39c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9z",
              ],
              "iconName": "cloud-arrow-down",
              "prefix": "fas",
            }
          }
          size={20}
        />
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(255, 255, 255)",
          "left": 0,
          "padding": 10,
          "position": "absolute",
        }
      }
    >
      <View
        accessibilityLabel="text translated"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="header.drawmenu"
      >
        <
          color="rgb(216, 216, 216)"
          icon={
            {
              "icon": [
                448,
                512,
                [
                  "navicon",
                ],
                "f0c9",
                "M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z",
              ],
              "iconName": "bars",
              "prefix": "fas",
            }
          }
          size={48}
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
          "padding": 15,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Text
        style={
          {
            "color": "rgb(216, 216, 216)",
            "fontSize": 8,
          }
        }
      >
        text translated
      </Text>
      <Text
        style={
          {
            "color": "rgb(216, 216, 216)",
            "fontSize": 8,
          }
        }
      >
        (undefinedxundefined)-undefined
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(0, 122, 255)",
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    accessibilityLabel="text translated"
    accessible={true}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={false}
        tintColor="rgb(28, 28, 30)"
        title="text translated"
      />
    }
    style={
      {
        "flexGrow": 1,
        "height": "100%",
        "marginTop": 10,
        "width": "100%",
      }
    }
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          {
            "display": "flex",
            "flexDirection": "column",
          }
        }
        testID="transactionList.1"
      >
        <View
          style={
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "borderBottomWidth": 1,
              "borderColor": "rgb(255, 255, 255)",
              "borderTopWidth": 1,
              "paddingBottom": 5,
              "paddingLeft": 15,
              "paddingTop": 5,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            Dec 2022
          </Text>
        </View>
        <View
          onPress={[Function]}
        >
          <View
            style={
              {
                "borderBottomColor": "rgb(216, 216, 216)",
                "borderBottomWidth": 1,
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 15,
                "paddingBottom": 15,
              }
            }
          >
            <
              color="rgb(0, 122, 255)"
              icon={
                {
                  "icon": [
                    384,
                    512,
                    [
                      8595,
                    ],
                    "f063",
                    "M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z",
                  ],
                  "iconName": "arrow-down",
                  "prefix": "fas",
                }
              }
              size={24}
              style={
                {
                  "marginLeft": 5,
                  "marginRight": 5,
                  "marginTop": 5,
                }
              }
            />
            <View
              style={
                {
                  "display": "flex",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 18,
                    "opacity": 0.65,
                  }
                }
              >
                Unknown
              </Text>
              <View
                style={
                  {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "opacity": 0.65,
                    }
                  }
                >
                  text translated
                </Text>
                <Text
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "opacity": 0.65,
                    }
                  }
                >
                  Dec 13, 8:00 am
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "baseline",
                  "alignSelf": "baseline",
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "justifyContent": "flex-end",
                  "paddingRight": 5,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "disabled": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "baseline",
                      "alignSelf": "baseline",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "flex-end",
                      "paddingRight": 5,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "rgb(0, 122, 255)",
                        "fontSize": 18,
                      }
                    }
                  >
                    á™‡
                  </Text>
                  <Text
                    style={
                      {
                        "color": "rgb(0, 122, 255)",
                        "fontSize": 18,
                        "fontWeight": "700",
                      }
                    }
                  >
                     0.1234
                  </Text>
                  <Text
                    style={
                      {
                        "color": "rgb(0, 122, 255)",
                        "fontSize": 12.6,
                        "paddingBottom": 0,
                      }
                    }
                  >
                    5678
                  </Text>
                  <Text
                    style={
                      {
                        "color": "rgb(0, 122, 255)",
                        "fontSize": 18,
                      }
                    }
                  >
                     á™‡
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "display": "flex",
            "flexDirection": "column",
          }
        }
        testID="transactionList.2"
      >
        <View
          onPress={[Function]}
        >
          <View
            style={
              {
                "borderBottomColor": "rgb(216, 216, 216)",
                "borderBottomWidth": 1,
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 15,
                "paddingBottom": 15,
              }
            }
          >
            <
              color="rgb(0, 122, 255)"
              icon={
                {
                  "icon": [
                    384,
                    512,
                    [
                      8595,
                    ],
                    "f063",
                    "M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z",
                  ],
                  "iconName": "arrow-down",
                  "prefix": "fas",
                }
              }
              size={24}
              style={
                {
                  "marginLeft": 5,
                  "marginRight": 5,
                  "marginTop": 5,
                }
              }
            />
            <View
              style={
                {
                  "display": "flex",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 18,
                    "opacity": 0.65,
                  }
                }
              >
                Unknown
              </Text>
              <View
                style={
                  {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "opacity": 0.65,
                    }
                  }
                >
                  text translated
                </Text>
                <Text
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "opacity": 0.65,
                    }
                  }
                >
                  Dec 13, 8:00 am
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "baseline",
                  "alignSelf": "baseline",
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "justifyContent": "flex-end",
                  "paddingRight": 5,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "disabled": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "baseline",
                      "alignSelf": "baseline",
                      "flexDirection": "row",
                      "flexGrow": 1,
                      "justifyContent": "flex-end",
                      "paddingRight": 5,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "rgb(0, 122, 255)",
                        "fontSize": 18,
                      }
                    }
                  >
                    á™‡
                  </Text>
                  <Text
                    style={
                      {
                        "color": "rgb(0, 122, 255)",
                        "fontSize": 18,
                        "fontWeight": "700",
                      }
                    }
                  >
                     0.8765
                  </Text>
                  <Text
                    style={
                      {
                        "color": "rgb(0, 122, 255)",
                        "fontSize": 12.6,
                        "paddingBottom": 0,
                      }
                    }
                  >
                    4321
                  </Text>
                  <Text
                    style={
                      {
                        "color": "rgb(0, 122, 255)",
                        "fontSize": 18,
                      }
                    }
                  >
                     á™‡
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "height": 150,
            "justifyContent": "flex-start",
            "marginBottom": 30,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(0, 122, 255)",
              "opacity": 0.65,
            }
          }
        >
          text translated
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
